### DISCUSSION TITLE
Welcome to zEpid Discussions!

### DISCUSSION THREAD

**Discussion by pzivich** [pzivich/zepid#153] [General]
Welcome to zEpid Discussions. I am going to try to use this as a way to post questions / ask for advice outside of the Issues tab

**Comment by miaow27** [comment]
@pzivich, thanks for prompt response on your github page. I am currently a new user of your package. Our team are actively using the `tmle` package developed in R, and I am looking for analternatives in Python. 

Want to share some feedback I have when comparing result using R's TMLE vs `zEpid` on a test data:
-  for parametric TMLE, both estimates and CI, P-value are very very close. 
-  for utilizing Machine Learning, 
  -  without using Singlecorssfit, the CI/P-value from `zEpid` is way too narrow
  -  However, by using Singlecrossfit (default setting), the CI/P-value is much similar (`zEpid` could be more conservative sometimes but conclusion are same as of using R)
  - It is quite interesting that even though the inference method between R and `SingleCrossFit` is different, it yields similar results. 
 
This is a great package that I hope you keep developing!

**Reply by pzivich** [reply]
Thanks @miaow27 ! Would you mind sharing what R TMLE library you are using? Also can you share the machine learning algorithms you are using? I am curious to know. Thanks!

**Reply by miaow27** [reply]
Hi @pzivich, I am using this R package developed by Susan Guber (https://www.jstatsoft.org/article/view/v051i13). 
I used your singlecrossfit TMLE for  the following scenario:
1) gmodel = glm, Q model = Superleaner(Randomforest, folds = 3)
2) gmodel = glm, Q model = Superleaner(Randomforest, glm, folds = 3)
3) gmodel = Superleaner(Randomforest, glm, folds = 3), Q model = Superleaner(Randomforest, glm, folds = 3)

with same setting in R. For example in scenario 2, q.Sl.library = c('SL.glm', 'SL.ranger'). 

Estimates and CI are close (not exactly same). My sample size is 2000 with 16% in treatment group.

**Reply by pzivich** [reply]
Thank you for sharing! Always good to know a bit about the applications

I would expect them to be close but there is a fair bit of randomness that occurs (results depend on how the data is exactly partitioned and seed used for algorithms like random forest, so exactly the same results is pretty unlikely across software).

**Comment by joannadiong** [comment]
Hi @pzivich, would there be plans in future to extend zEpid's IPTW functionality to support continuous exposures? 
Right now, [this post](https://www.andrewheiss.com/blog/2020/12/01/ipw-binary-continuous/#ipw-with-the-ipw-package-continuous-treatment) suggests performing IPW for continuous exposures using the `ipw` or `WeightIt` packages in R.
Thanks!

**Reply by pzivich** [reply]
Hi @joannadiong 

So, I have thought about it and it is a functionality that would be good to have. The difficulty is there are lots of distributions / regression models one can choose with a continuous exposure. The following paper gets into the pieces here a bit more https://pubmed.ncbi.nlm.nih.gov/24487212/

Broadly, the weights look like 1/f(A=a|W) where f is the density function. The exact distribution for the density function needs to be updated based on the regression model chosen. The main problem is that it makes custom models difficult to handle (I don't know if I can 'detect' what distribution the model is using). 

Basically the hurdle I have to overcome is the generality of the f(A=a|W) piece. The continuous exposure piece is under the hood for my other library, MossSpider. So, I may be able to adapt something

**Reply by joannadiong** [reply]
Hi @pzivich, 

Thanks for this. I appreciate your considering the complexities and am pleased that adding this functionality could be a work in progress. I think it would be highly valuable.

From an applied perspective, many research questions in the public health and neuroscience areas I work across involve continuous exposures. One approach that VanderWeele recommends is to binarise a continuous exposure (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3178663/). This approach arguably generates a more precise causal question, but inevitably coarsens the exposure.

The other approach would be to model the continuous exposure using IPW and stabilised weights.

If I had to model continuous exposures, I could make do with those R packages in interim. But having a Python alternative for the longer term would be ideal. Thanks again!
